Src/AutoFixture.NUnit3/AutoDataAttribute.cs

```diff
@@ -0,0 +1,275 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+using NUnit.Framework.Internal.Builders;
+using Ploeh.AutoFixture.Kernel;
+using ITestCaseData = NUnit.Framework.Interfaces.ITestCaseData;
+
+namespace Ploeh.AutoFixture.NUnit3
+{
+    /// <summary>
+    /// This attribute uses AutoFixture to generate values for unit test parameters. 
+    /// This implementation is based on TestCaseAttribute of NUnit3
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method)]
+    public class AutoDataAttribute : NUnitAttribute, ITestBuilder, ITestCaseData, IImplyFixture
+    {
+        private readonly IFixture _fixture;
+
+        /// <summary>
+        /// Construct a <see cref="AutoDataAttribute"/>
+        /// </summary>
+        public AutoDataAttribute()
+            : this(new Fixture())
+        {
+        }
+
+        /// <summary>
+        /// Construct a <see cref="AutoDataAttribute"/> with an <see cref="IFixture"/>
+        /// </summary>
+        protected AutoDataAttribute(IFixture fixture)
+        {
+            _fixture = fixture;
+            this.RunState = RunState.Runnable;
+            this.Properties = new PropertyBag();
+        }
```

Comment:
Does this constructor need to be here?

If I delete it, and slightly modify the parameterless constructor to the following implementation, the build still succeeds.

```C#
public AutoDataAttribute()
{
    this._fixture = new Fixture();
}
```